db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+
+`<db:sql>SELECT price\:\:float8 FROM PRODUCT</db:sql>`
+====
+
+[NOTE]
+====
+*Since version 1.8.0 (SQL Casting for PostgreSQL and Snowflake)*:
+Database Connector now accepts SQL Casting PostgreSQL and Snowflake syntax's, through the double colon (::) expression without the need to escape each colon (this feature doesn't affect the escaping colons behaviour). For example:
+
+`<db:sql>SELECT MAX(modified_date)::DATE FROM sales</db:sql>`
+====
+
+
+== Configure Dynamic Queries
+
+When you need to parameterize not only the `WHERE` clause, but also parts of the query itself, for example queries that need to compare tables that depend on a condition, or complex queries for which the project table columns need to vary, you can configure dynamic queries.
+
+In the following example, you configure a dynamic query by using a full expression with a string in which the table depends on a variable `$(vars.table)`. Although, some of the query text is dynamic (`"SELECT * FROM $(vars.table)`), the `WHERE` clause still defines the `WHERE` condition using input parameters, in this case, `WHERE name = :name`.
+
+. In your Studio flow, select the *Select* operation.
+. In the operation configuration screen, set the *SQL Query Text* field to `SELECT * FROM $(vars.table) WHERE name = :name`.
+. Set the *Input Parameters* field to `{'name' : payload}`.
+
+The following screenshot shows the dynamic query configuration:
+
+.Dynamic query configuration
+image::database-select-operation-3.png[Dynamic query configuration configuration in Studio]
+
+In the XML editor, the `<db:sql>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<set-variable variableName="table" value="PLANET"/>
+<db:select config-ref="dbConfig">
+    <db:sql>#["SELECT * FROM $(vars.table) WHERE name = :name"]</db:sql>
+    <db:input-parameters>
+        #[{'name' : payload}]
+    </db:input-parameters>
+</db:select>
+----
+
+Input parameters can be applied only to parameters in a `WHERE` clause. To modify any other part of the query, use the DataWeave’s interpolation operator.
+
+
+== Stream Large Results
+
+Use streaming with queries that return many records, such as in integration use cases. In Mule 4, streaming is transparent and always enabled.
+
+For example, if you submit a query that returns 10K rows by attempting to fetch all those rows at once results in both performance degradation, due to the big pull from the network, and the risk of running out of memory, because all the information must be loaded into RAM.
+
+With streaming, Database Connector fetches and processes only part of the query at one time, reducing the load on the network and memory.  This means that the connector does not fetch the 10K rows at once; instead, it fetches a smaller chunk, and once that chunk is consumed, it fetches the rest.
+
+You can also use the new repeatable streams mechanism where DataWeave and other components process the same stream many times, even in parallel.
```

Comment:
Maybe avoid "new," because at some point, this is not going to be new?
```suggestion
You can also use the repeatable streams mechanism, which enables DataWeave and other components to process the same stream many times, even in parallel.
```